x/wasm/keeper/keeper_extension.go

```diff
@@ -0,0 +1,66 @@
+package keeper
+
+import (
+	sdk "github.com/line/lbm-sdk/types"
+	sdkerrors "github.com/line/lbm-sdk/types/errors"
+	wasmtypes "github.com/line/lbm-sdk/x/wasm/types"
+)
+
+func (k Keeper) IsInactiveContract(ctx sdk.Context, contractAddress sdk.AccAddress) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(wasmtypes.GetInactiveContractKey(contractAddress))
+}
+
+func (k Keeper) IterateInactiveContracts(ctx sdk.Context, fn func(contractAddress sdk.AccAddress) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+	prefix := wasmtypes.InactiveContractPrefix
+	iterator := sdk.KVStorePrefixIterator(store, prefix)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		contractAddress := sdk.AccAddress(iterator.Value())
+		if stop := fn(contractAddress); stop {
+			break
+		}
+	}
+}
+
+func (k Keeper) addInactiveContract(ctx sdk.Context, contractAddress sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := wasmtypes.GetInactiveContractKey(contractAddress)
+
+	store.Set(key, contractAddress)
+}
+
+func (k Keeper) deleteInactiveContract(ctx sdk.Context, contractAddress sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := wasmtypes.GetInactiveContractKey(contractAddress)
+	store.Delete(key)
+}
+
+// activateContract delete the contract address from inactivateContract list if the contract is deactivated.
+func (k Keeper) activateContract(ctx sdk.Context, contractAddress sdk.AccAddress) error {
+	if !k.IsInactiveContract(ctx, contractAddress) {
+		return sdkerrors.Wrapf(wasmtypes.ErrNotFound, "no inactivate contract %s", contractAddress.String())
+	}
+
+	k.deleteInactiveContract(ctx, contractAddress)
+	k.bank.DeleteFromInactiveAddr(ctx, contractAddress)
+
+	return nil
+}
+
+// deactivateContract add the contract address to inactivateContract list.
+func (k Keeper) deactivateContract(ctx sdk.Context, contractAddress sdk.AccAddress) error {
+	if k.IsInactiveContract(ctx, contractAddress) {
+		return sdkerrors.Wrapf(wasmtypes.ErrAccountExists, "already inactivate contract %s", contractAddress.String())
+	}
+	if !k.HasContractInfo(ctx, contractAddress) {
+		return sdkerrors.Wrapf(wasmtypes.ErrInvalid, "no contract %s", contractAddress.String())
```

Comment:
ErrInvalid -> ErrNotFound ?

Reply:
In this case, the `contractAddress` must be the address of contract, but it was confirmed it is not a contract address throught `HasContractInfo`. So I wanted to notice this is **invalid** request.

Reply:
Isn't HasContractInfo simply a function that checks whether the address is registered in the store, not whether it is the contract's address?

Reply:
I wanted to check a contract address. and I used it considering it as a function that `HasContractInfo` can check. Is there any other function that can check if it is a contract address or not?

Reply:
Cosmwasm does not distinguish between contract address and account address, so I don't think we can check which value the address is.
HasContractInfo is a function that checks if the address is an installed contract, so no matter what value is entered, if the corresponding ContractInfo does not exist, the error is not Found.

Reply:
Good. It is correct as an error of `HasContractInfo`, but it is not appropriate as an error of `deactivateContract`.